chalice/deploy/deployer.py

```diff
@@ -59,6 +59,7 @@ def validate_configuration(config):
     """
     routes = config.chalice_app.routes
     validate_routes(routes)
+    validate_route_content_types(routes, config.chalice_app.api.binary_types)
```

Comment:
Yeah that would make sense to move these out to subclasses in the future given we are probably only going to be adding more validations. But it does not need to be addressed in this PR.